tests/test_vqe_vqemcscf.py

```diff
@@ -42,31 +45,52 @@ def test_vqecasci_h2o_2e_2o():
 
 
 def test_ssvqecasci_h2o_2e_2o():
+
+    n_electron = 2
+    n_orbitals = 2
+    excitation_number = 5
+
+    # Generate inital states
+    inital_states = _generate_inital_states(n_electron, n_orbitals, excitation_number)
+    state_2 = inital_states[2]
+    state_3 = inital_states[3]
+
+    # Define spin-adapted linear combinations
+    state_2_spin_adapted = comp_basis_superposition(
+        state_2, state_3, theta=+pi / 4, phi=0.0
+    )
+    state_3_spin_adapted = comp_basis_superposition(
+        state_2, state_3, theta=-pi / 4, phi=0.0
+    )
+
+    # Replace initial states with spin-adapted states. Needed as Ansatz is spin conserving
+    inital_states[2] = state_2_spin_adapted
+    inital_states[3] = state_3_spin_adapted
+
     mc = vqemcscf.VQECASCI(
         mf,
-        2,
-        2,
+        n_electron,
+        n_orbitals,
         optimizer=LBFGS(),
+        inital_states=inital_states,
         ansatz=Ansatz.GateFabric,
         layers=2,
-        excitation_number=3,
+        excitation_number=inital_states,
     )
+
+    # Create reference CASCI values
     mc.kernel()
+    refmc = mcscf.CASCI(mf, n_orbitals, n_electron)
+    refmc.fcisolver.nroots = excitation_number
+    refmc.kernel()
 
-    # refmc = mcscf.CASCI(mf, 2, 2)
-    # refmc.fcisolver.nroots = 4
-    # refmc.kernel()
-    # print(refmc.e_tot)
-
-    ref_energies = [
-        -74.96569511044997,
-        -74.56710399202521,
-        -74.52886408322473,
-        -74.52886408322476,
-    ]
-    assert all(
-        utils.almost_equal(a, b) for a, b in zip(mc.fcisolver.energies, ref_energies)
-    )
+    def all_elements_within_threshold(array1, array2, threshold=10**-7):
```

Comment:
I'm interested in the numerical results of the current test with the changed initial state. While you’re making other corrections, could you comment them out and paste the numerical values in as comments?